dist/common/scripts/scylla_util.py

```diff
@@ -154,20 +154,27 @@ def __instance_metadata(self, path, recursive=False):
         return curl(self.META_DATA_BASE_URL + path + "?recursive=%s" % str(recursive).lower(),
                     headers={"Metadata-Flavor": "Google"})
 
+    def is_in_root_devs(self, x, root_devs):
+        for root_dev in root_devs:
+            if root_dev.startswith(os.path.join("/dev/", x)):
+                return True
+        return False
+
     def _non_root_nvmes(self):
         """get list of nvme disks from os, filter away if one of them is root"""
         nvme_re = re.compile(r"nvme\d+n\d+$")
 
         root_dev_candidates = [x for x in psutil.disk_partitions() if x.mountpoint == "/"]
-        if len(root_dev_candidates) != 1:
-            raise Exception("found more than one disk mounted at root ".format(root_dev_candidates))
+        #it can be empty too - e.g. in k8s
+        #if len(root_dev_candidates) != 1:
+        #    raise Exception("found more than one disk mounted at root ".format(root_dev_candidates))
 
-        root_dev = root_dev_candidates[0].device
+        root_devs = [x.device for x in root_dev_candidates]
         # if root_dev.startswith("/dev/mapper"):
         #     raise Exception("mapper used for root, not checking if nvme is used ".format(root_dev))
 
-        nvmes_present = list(filter(nvme_re.match, os.listdir("/dev")))
-        return {self.ROOT: [root_dev], self.EPHEMERAL: [x for x in nvmes_present if not root_dev.startswith(os.path.join("/dev/", x))]}
+        nvmes_present = list(filter(nvme_re.match, os.listdir("/dev")))   # it can be empty too - e.g. in k8s
+        return {self.ROOT: [root_devs], self.EPHEMERAL: [x for x in nvmes_present if not self.is_in_root_devs(x, root_devs)]}
```

Comment:
Is this related to the patch subject?

Reply:
definitely, the logic changed, now for every root device we need to check if it belongs to an nvme and filter that nvme out from available ones

reason is simple - overlay mounts in K8S create multiple / mounts